micro-rdk-server/esp32/main.rs

```diff
@@ -101,30 +100,35 @@ mod esp32 {
                     .into(),
                 )
                 .expect("Failed to store robot credentials to NVS");
-
-            serve_web(None, repr, max_connection, storage);
         }
-        #[cfg(not(has_robot_config))]
-        {
-            use micro_rdk::common::provisioning::server::ProvisioningInfo;
+
+        let max_connections = unsafe {
+            if !g_spiram_ok {
+                log::info!("spiram not initialized disabling cache feature of the wifi driver");
+                g_wifi_feature_caps &= !(CONFIG_FEATURE_CACHE_TX_BUF_BIT as u64);
+                1
+            } else {
+                3
+            }
+        };
+
+        let info = if cfg!(feature = "provisioning") {
             let mut info = ProvisioningInfo::default();
             info.set_manufacturer("viam".to_owned());
             info.set_model("test-esp32".to_owned());
-            let storage = NVSStorage::new("nvs").unwrap();
-            serve_web(Some(info), repr, max_connection, storage);
+            Some(info)
+        } else {
+            None
+        };
+
+        if info.is_none() && !storage.has_wifi_credentials() {
```

Comment:
just clarifying, when we hit this scenario, the only recourse is to flash new firmware right?

Reply:
correct, if there's no way for the device to provision itself and there's no wifi credentials then it's isolated. adding another log line with info on what suggested action is.